### PR TITLE
Add promdb prometheus helper

### PR DESCRIPTION
**Pull request by mikemrm**

### CODE REVIEW

File: prometheus/promdb/README.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# PromDB
+A quick library for adding prometheus metrics to your service for your databases
+
+## Usage
+
+### Without labels
+
+```golang
```
Comment (mmlb) [inline prometheus/promdb/README.md:8]: these should be `go`, not `golang`

File: prometheus/promdb/promdb.go [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+// WithLabel is a shortcut to With for providing a single label and value
+func (c *DatabaseCollector) WithLabel(name, value string) *DatabaseCollector {
+	labels := prometheus.Labels{name: value}
+	return c.With(labels)
+}
+
+// Describe implements prometheus Collector
+func (c *DatabaseCollector) Describe(ch chan<- *prometheus.Desc) {
```
Comment (mmlb) [inline prometheus/promdb/promdb.go:52]: I assume this pattern was figured out from a prometheus doc? Would be good to link to it to know where this non-obvious-send-pattern comes from (in the commit message maybe, but could be in line too)

File: prometheus/promdb/promdb.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
+package promdb
```
Comment (mmlb) [inline prometheus/promdb/promdb.go:1]: we should have package level docs in a doc.go file

File: prometheus/promdb/promdb.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+	ch <- prometheus.MustNewConstMetric(c.inUse, prometheus.GaugeValue, inUse)
+	ch <- prometheus.MustNewConstMetric(c.idle, prometheus.GaugeValue, idle)
+	ch <- prometheus.MustNewConstMetric(c.waitCount, prometheus.GaugeValue, waitCount)
+	ch <- prometheus.MustNewConstMetric(c.waitDuration, prometheus.GaugeValue, waitDuration)
+	ch <- prometheus.MustNewConstMetric(c.maxIdleClosed, prometheus.GaugeValue, maxIdleClosed)
+	ch <- prometheus.MustNewConstMetric(c.maxLifetimeClosed, prometheus.GaugeValue, maxLifetimeClosed)
+}
+
+// NewDatabaseCollector creates a new instance of DatabaseCollector with no
+// labels
```
Comment (mmlb) [inline prometheus/promdb/promdb.go:87]: just one line

File: prometheus/promdb/promdb.go [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+func NewDatabaseCollector(db DBWithStats) *DatabaseCollector {
+	return NewDatabaseCollectorWithLabels(db, nil)
+}
+
+// NewDatabaseCollectorWithLabels creates a new instance of DatabaseCollector
+// with the specified prometheus labels
```
Comment (mmlb) [inline prometheus/promdb/promdb.go:93]: ditto

File: prometheus/promdb/promdb.go [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+	inUse              *prometheus.Desc
+	idle               *prometheus.Desc
+	waitCount          *prometheus.Desc
+	waitDuration       *prometheus.Desc
+	maxIdleClosed      *prometheus.Desc
+	maxLifetimeClosed  *prometheus.Desc
+}
+
+// With returns a new instance of DatabaseCollector with the specified labels
+// merging with any previous labels that may have been defined
```
Comment (mmlb) [inline prometheus/promdb/promdb.go:30]: just one line

File: prometheus/promdb/promdb_test.go [hunk 7]
```diff
@@ -0,0 +1,241 @@
+package promdb
+
+import (
+	"database/sql"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
```
Comment (mmlb) [inline prometheus/promdb/promdb_test.go:10]: dto?

File: prometheus/promdb/README.md (hunk 1 above)
Comment (mikemrm) [reply prometheus/promdb/README.md:8]: ah, thx

File: prometheus/promdb/promdb.go (hunk 3 above)
Comment (mikemrm) [reply prometheus/promdb/promdb.go:1]: i'll have to look into that

File: prometheus/promdb/promdb_test.go (hunk 7 above)
Comment (mikemrm) [reply prometheus/promdb/promdb_test.go:10]: this import was just pulled from one of their files. I can rename this to something. The actual package name is `io_prometheus_client` I can rename it to promclient or something along those lines.

File: prometheus/promdb/promdb.go (hunk 2 above)
Comment (mikemrm) [reply prometheus/promdb/promdb.go:52]: I've added some additional details (that it's implementing an interface) but I think this should be relatively obvious that you can find it in their docs.

File: prometheus/promdb/promdb.go (hunk 3 above)
Comment (mikemrm) [reply prometheus/promdb/promdb.go:1]: added

File: prometheus/promdb/promdb_test.go (hunk 7 above)
Comment (mikemrm) [reply prometheus/promdb/promdb_test.go]: renamed to promproto

File: prometheus/promdb/promdb.go (hunk 6 above)
Comment (mmlb) [reply prometheus/promdb/promdb.go:30]: ... still not just one line :qq:

File: prometheus/promdb/doc.go [hunk 8]
```diff
@@ -0,0 +1,35 @@
+/*
+Package promdb is a quick library for adding prometheus metrics to your service for your databases,
+utilizing the sql.DBStats struct results.
```
Comment (mmlb) [inline prometheus/promdb/doc.go:3]: ditto, one line

**Pull request merged**